sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -43,7 +43,7 @@ abstract class DataType {
   /**
    * Enables matching against DataType for expressions:
    * {{{
-   *   case Cast(child @ BinaryType(), StringType) =>
+   *   case Cast(child @ DataType(), StringType) =>
```

Comment:
I'm not sure we actually want to make this change.  This is just an illustration of a case where this function would be used.  Matching on `DataType` itself is not possible, right?